### PR TITLE
Add testing sidebars (removes playground)

### PR DESCRIPTION
**Pull request by amiraliemami**
Reviewer: @sidravi1 
Estimate: 1hr

---

## Ticket

Fixes: [740](https://idinsight.atlassian.net/browse/AAQ-740)

### Goal

- add test sidepanels on the QA and UD pages as per new figma design.

### Changes

- Added sidebars
- Removed playground
- Updated NavBar
- Docs updates

### Future Tasks (optional)

- Cleanup alignment on the page
- Remove feedback from the sidebar

## How has this been tested?

- dev
- docker compose

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated affected documentation
- [ ] I have added a blogpost in Latest Updates

### CODE REVIEW

File: admin_app/src/app/urgency-rules/page.tsx [hunk 1]
```diff
@@ -142,160 +139,219 @@ const UrgencyRulesPage = () => {
...
+              sx={{
+                bgcolor: "orange",
+                position: "absolute",
+                bottom: 80,
+                right: 150,
+              }}
+              onClick={handleSidebarToggle}
+            >
+              <PlayArrow />
+              <Layout.Spacer horizontal multiplier={0.3} />
```
Comment (sidravi1) [inline admin_app/src/app/urgency-rules/page.tsx:336]: The positioning of this button looks odd.

image

Would be nice to have it at the bottom of ths screen but aligned under the "New" button.

File: admin_app/src/app/urgency-rules/UDSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
+import React, { useState } from "react";
```
Comment (sidravi1) [inline admin_app/src/app/urgency-rules/UDSidebar.tsx:1]: We have been putting components under `/component`. So suggest moving this to `urgency-rules/components`

File: admin_app/src/app/content/SearchSidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,388 @@
+import React, { useState } from "react";
```
Comment (sidravi1) [inline admin_app/src/app/content/SearchSidebar.tsx:1]: We have been putting components under /component. So suggest moving this to `content/components`

File: docs/components/admin-app/playground.md [hunk 4]
```diff
@@ -1,8 +0,0 @@
-# Playground
```
Comment (sidravi1) [inline docs/components/admin-app/playground.md:1]: I get a broken link in the blogs. How should we handle this? 

```
WARNING -  Doc file 'blog/posts/playground.md' contains a relative link
           '../../components/admin-app/playground.md', but the target 'components/admin-app/playground.md'
           is not found among documentation files.
```

File: admin_app/src/components/TypingAnimation.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { keyframes } from "@emotion/react";
```
Comment (sidravi1) [inline admin_app/src/components/TypingAnimation.tsx:1]: This is also under `@/components/Common`

And is used under `DetailsDrawer.tsx` under performance

File: admin_app/src/components/SidebarCommon.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  CircularProgress,
+  FormControlLabel,
+  IconButton,
+  Paper,
+  Typography,
+  TextField,
+} from "@mui/material";
+
+import { useAuth } from "@/utils/auth";
+import { SearchResponseBoxData } from "../app/content/SearchSidebar";
```
Comment (sidravi1) [inline admin_app/src/components/SidebarCommon.tsx:16]: We shouldn't have low-level components importing from the routes components. One option is to declare a higher level type and `SearchResponseBoxData` and `UDResponseBoxData` inherit / extend it.

File: admin_app/src/app/urgency-rules/UDSidebar.tsx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+              </Typography>
+            </Box>
+          </Fade>
+        </Modal>
+      </>
+    );
+  }
+};
+
+const UDSidebar = ({ closeSidebar }: { closeSidebar: () => void }) => {
```
Comment (sidravi1) [inline admin_app/src/app/urgency-rules/UDSidebar.tsx:206]: This doesn't extend all the way to the bottom for some reason. It seems to do so for the `SearchSidebar`. 

image

Is this easy to fix?

**Review by sidravi1 — APPROVED** [review]
Thanks @amiraliemami! It works well.

Since we only have a few days left, I don't want to be a blocker for design things. 

Overall, I think we should try to use an existing component like Drawer and then modify behaviour instead of making it from scratch. I'm happy to take a crack at this once I am done with page 3 with Mark.

File: admin_app/src/components/TypingAnimation.tsx (hunk 5 above)
Comment (amiraliemami) [reply admin_app/src/components/TypingAnimation.tsx:1]: ohh I didnt know that - it was part of playground so I pulled it out. Will remove

**Pull request merged**